runtime/src/message_processor.rs

```diff
@@ -214,24 +194,63 @@ impl<'a> InvokeContext for ThisInvokeContext<'a> {
     fn verify_and_update(
         &mut self,
         instruction: &CompiledInstruction,
-        accounts: &[(Pubkey, Rc<RefCell<AccountSharedData>>)],
+        account_indices: &[usize],
         write_privileges: &[bool],
     ) -> Result<(), InstructionError> {
         let stack_frame = self
             .invoke_stack
             .last()
             .ok_or(InstructionError::CallDepth)?;
+        let program_id = &stack_frame.key;
+        let rent = &self.rent;
         let logger = self.get_logger();
-        InstructionProcessor::verify_and_update(
-            instruction,
-            &mut self.pre_accounts,
-            accounts,
-            &stack_frame.key,
-            &self.rent,
-            write_privileges,
-            &mut self.timings,
-            logger,
-        )
+        let accounts = &self.accounts;
+        let pre_accounts = &mut self.pre_accounts;
+        let timings = &mut self.timings;
+
+        // Verify the per-account instruction results
+        let (mut pre_sum, mut post_sum) = (0_u128, 0_u128);
+        let mut work = |_unique_index: usize, index_in_instruction: usize| {
+            if index_in_instruction < write_privileges.len()
+                && index_in_instruction < accounts.len()
```

Comment:
```suggestion
                && index_in_instruction < account_indices.len()
```
I think this should be updated

Reply:
Good find!